Framework/Geometry/src/Objects/Object.cpp

```diff
@@ -1120,10 +1113,10 @@ double Object::triangleSolidAngle(const V3D &observer,
     //
     return rayTraceSolidAngle(observer); // so is this
   }
-  double *vertices = this->getTriangleVertices();
-  int *faces = this->getTriangleFaces();
+  const auto &vertices = this->getTriangleVertices().get();
```

Comment:
You have gone through the trouble of making this `boost::optional` but you don't need to check that it is valid before extracting?